### PR TITLE
Improve user auth

### PR DESCRIPTION
**Pull request by bongochin**
Changes:
- fixed error message not displaying when user enters invalid username
- fixed error message not displaying when user enters invalid password
- improved error message to display currently logged in user
- fixed error message displaying when signup is completed successfully
- fixed erroneous login fail falling into loop

### CODE REVIEW

**Review by chrisreadsf — APPROVED** [review]

**Comment by bongochin** [comment]
Derek - Please review the branch conflit on index.pug, thank you!

File: routes/index.js [hunk 1]
```diff
@@ -5,9 +5,20 @@ const db = require('../db/models');
 
 /* GET home page. */
 router.get('/', function(req, res, next) {
-  res.render('index', { title: 'GoodLaughs' });
+  let username = [];
```
Comment (ghost) [inline routes/index.js]: I'm curious why this is an array. I'm assuming you have some sort of possibility in your template of rendering out multiple usernames.

File: routes/users.js [hunk 2]
```diff
@@ -113,7 +127,7 @@ router.post('/signup', csrfProtection, validateSignup, handleValidationErrors, a
 
   const validatorErrors = validationResult(req);
   if (validatorErrors.isEmpty()) {
-    const user = db.User.build({
+    const user = await db.User.build({
```
Comment (ghost) [inline routes/users.js:130]: `.build` is not an asynchronous operation. Usually you will do `.build` to create an instance in your server program, and then `.save` to commit that instance as a row into your database. `.save` WILL be asynchronous, and should be used with the `await` operator; using `await` with `.build` doesn't actually do anything.

File: routes/users.js [hunk 3]
```diff
@@ -134,56 +148,18 @@ router.post('/signup', csrfProtection, validateSignup, handleValidationErrors, a
         username: user.username,
         id: user.id,
       };
-      req.session.save(() => {
-        res.redirect('/');
-      })
+      req.session.save()
+      res.redirect('/');
```
Comment (ghost) [inline routes/users.js]: You want lines 151 and 152 to be replaced with:
```js
req.session.save(err => {
  if (err) return next(err); // Pass along any possible error in committing the session to the database to your error handling middleware.
  res.redirect("/"); // It is now safe to redirect and ensure that your session has been saved.
});
```

**Review by ghost — APPROVED** [review]
Not much to review here, and it LGTM (looks good to me 😄). I left a few comments about things that stood out to me. Great job!

**Review by dereknungesser — APPROVED** [review]

**Pull request merged**